sdk/core/azure-core/test/ut/transport_adapter.cpp

```diff
@@ -439,4 +439,32 @@ namespace Azure { namespace Core { namespace Test {
     CheckBodyFromBuffer(*response, expectedResponseBodySize);
   }
 
+  TEST_F(TransportAdapter, requestFailedException)
+  {
+    Azure::Core::Http::Url host("http://unresolvedHost.org/get");
+
+    auto request = Azure::Core::Http::Request(Azure::Core::Http::HttpMethod::Get, host);
+    EXPECT_THROW(pipeline.Send(context, request), Azure::Core::RequestFailedException);
+  }
+
+  TEST_F(TransportAdapter, dynamicCast)
+  {
+    Azure::Core::Http::Url host("http://unresolvedHost.org/get");
+    auto request = Azure::Core::Http::Request(Azure::Core::Http::HttpMethod::Get, host);
+
+    // test dynamic cast
+    try
+    {
+      auto result = pipeline.Send(context, request);
+    }
+    catch (Azure::Core::RequestFailedException& err)
+    {
+      std::cout << err.what();
```

Comment:
Why are we printing to stdout in a unit test? Either remove, or add an assert check that compares the string message (though that isn't needed, since we can change the message string).
```suggestion
```